docs/NNNN-mutable-head.md

```diff
@@ -0,0 +1,287 @@
+2. Create a new revision directory somewhere in temp space named `rN`.
+3. Create a copy of the existing mutable HEAD inventory, hence forth referred to as the "new inventory".
+4. Write any files that are added with digests not in the mutable HEAD inventory's manifest to the temp revision directory, and add corresponding manifest entries to the new inventory.
+5. Update the version state in the new inventory to reflect any changes that were made to the object (deletions, renames, additions, modifications).
+6. Update the new inventory’s manifest and fixity blocks, removing any entries that were introduced in prior revisions of the mutable HEAD but are no longer referenced in the current version state.
+7. Write the new inventory and its sidecar to temp space.
+8. Write the file `[object-root]/extensions/NNNN-mutable-head/revisions/rN` containing the text `rN`. If this fails, abort because another process already created a revision using the same revision number.
+9. Move the temp revision directory to `[object-root]/extensions/NNNN-mutable-head/head/content/rN`.
+10. Move the new inventory file and sidecar into `[object-root]/extensions/NNNN-mutable-head/head`.
+11. Finally, iterate over all of the files under `[object-root]/extensions/NNNN-mutable-head/head/content`, deleting any that are no longer referenced in the inventory manifest.
+
+### Committing a Mutable HEAD
+
+A commit is simply the process of moving an object’s mutable HEAD into the object’s root as an immutable OCFL version. Because the mutable HEAD is already a valid OCFL version, this is relatively straightforward.
+
+1. Compare the contents of `[object-root]/extensions/NNNN-mutable-head/root-inventory.json.sha512` and `[object-root]/inventory.json.sha512` to ensure that they are the same and that the root object has not been modified since the creation of the mutable HEAD. If their contents are different, there is a version conflict that is up to the implementation to resolve.
+2. Rewrite the inventory manifest and fixity entries that reference `extensions/NNNN-mutable-head/head` to reference `vN` instead, where `vN` is the mutable HEAD version number.
+3. Move `[object-root]/extensions/NNNN-mutable-head/head` to `[object-root]/vN`.
+4. If the directory cannot be moved because there is already a version directory at that location, then there is a version conflict that is up to the implementation to resolve.
+5. Write the updated inventory and sidecar to the object root.
+6. If this fails, move `[object-root]/vN` back to `[object-root]/extensions/NNNN-mutable-head/head` and abort.
+7. Otherwise, write the updated inventory and sidecar to `[object-root]/vN`.
+8. Remove the extension directory.
+
+The following is what the [mutable HEAD inventory example](#inventory) would look like after it is committed:
+
+```json
+{
+  "digestAlgorithm": "sha512",
+  "head": "v2",
+  "id": "ark:/12345/bcd987",
+  "manifest": {
+    "4d27c8...b53": [ "v2/content/r1/foo/bar.xml" ],
```

Comment:
The `r1`, `r2` and `r3` paths should be removed, no?

Reply:
No. The version content in the mutable head is _not_ rewritten when it is moved into the object root.

So, entries like `extensions/NNNN-mutable-head/head/content/r1/foo/bar.xml` are rewritten in the manifest to look like `vN/content/r1/foo/bar.xml`. Files within the content directory do not change location; only the content directory itself moves.

Reply:
Seems that one could choose to do it either way. Keeping the `rN` does simplify the content move

Reply:
Yes, this is true. It is an implementation detail. It is much more expensive to safely create the immutable version if you want to rewrite the content directory, but there is not that precludes doing so.

Reply:
I think this is resolved by the discussion above in concert with the opening comment of the section _"A commit is simply the process of moving an object’s mutable HEAD into the object’s root as an immutable OCFL version."_ -- that will preserve the `rN` directories

Reply:
Would it make sense to add a note highlighting this detail? and possibly emphasizing the fact that the OCFL allows for separating the notions of "Content Paths" and "Logical Paths"?